mantidproject/mantid
# LET data in data analysis

**Description of work**
This is to be merged after https://github.com/mantidproject/mantid/pull/35655

This PR is the second part of the above work, with both of these LET will be able to use the convfit tab in data analysis. At present this is not possible because both the sample data (fixed by #35655) and the resolution data (fixed by this) contains NANs. The NAN's in the resolution data caused problems as they were being integrated when the convolution was evaluated as part of the fit. The solution is to replace them with zeros (they are normally at the ends of the data so should be zero) and to warn the user. 


Ross


**To test:**

Email me for some data, I will send 4 files. 
Open the interface: indirect -> data analysis
Go to the convfit tab
Press `Add Workspace`
Select the IRIS data for the workspace and resolution (ends in _res)
In workspace indices put `0-5`
Press Add, then close the pop up
Change the number of Lorentzians to 1
Change the background to linearBackground
Press `Run`, a fit will appear on the plot

Click the StartX column header, then press remove
The Data input table will now be empty
Press `Add Workspace`
Select the LET data for the workspace (93338) and resolution (93343)
In workspace indices put `0-5`
Press Add, a warning will appear - close it. 
Close the pop up for data selection
In the fit property table, tick `Ignore invalid data`
Press Run - it will not crash!


Fixes #35716. 






---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
